api/nfd/v1alpha1/types.go

```diff
@@ -296,6 +299,34 @@ type MatchExpression struct {
 	// In other cases Value should contain at least one element.
 	// +optional
 	Value MatchValue `json:"value,omitempty"`
+
+	// IsMatch indicates whether the processed expression matches the host.
+	// This field should not be exposed in the CRD or used by users.
+	// It is for informational purposes only.
+	IsMatch bool `json:"-"`
+}
+
+// Override MarshalJSON to hide the IsMatch field,
```

Comment:
I think we want to start the doc-comment with the func/method name `MarshalJSON`

Reply:
Fixed.